bears/documentation/DocumentationStyleBear.py

```diff
@@ -0,0 +1,77 @@
+    DocumentationComment)
+from coalib.bearlib.languages.documentation.DocstyleDefinition import (
+    DocstyleDefinition)
+from coalib.bearlib.languages.documentation.DocumentationExtraction import (
+    extract_documentation)
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+
+
+class DocumentationStyleBear(LocalBear):
+    LANGUAGES = {language for docstyle, language in
+                 DocstyleDefinition.get_available_definitions()}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    ASCIINEMA_URL = 'https://asciinema.org/a/7sfk3i9oxs1ixg2ncsu3pym0u'
+    CAN_DETECT = {'Documentation'}
+
+    def run(self, filename, file, docstyle: str, language: str):
+        """
+        Checks for certain in-code documentation styles.
+
+        It currently checks for the following style:
+
+        ::
+
+            hello
+            :param x:
+                4 space indent
+            :return:
+                also 4 space indent
+                following lines are also 4 space indented
+        """
+        for doc_comment in extract_documentation(file, language, docstyle):
+            metadata = iter(doc_comment.parse())
+
+            # Assuming that the first element is always the only description.
+            new_metadata = [next(metadata)]
+
+            for m in metadata:
+                stripped_desc = list(map(str.lstrip, m.desc.splitlines(True)))
+                stripped_desc[0] = (('\n' + ' ' * 4 + stripped_desc[0])
+                                    if not stripped_desc[0] == ''
+                                    else stripped_desc[0])
+                new_desc = ''.join(('\n' if not line.endswith('\n') else '') +
+                                   (' ' * 4 if line != '' else '') + line
+                                   for line in stripped_desc)
+                new_metadata.append(m._replace(desc=new_desc))
+
+            new_comment = DocumentationComment.from_metadata(
+                new_metadata, doc_comment.docstyle_definition,
+                doc_comment.marker, doc_comment.indent, doc_comment.range)
+
+            if new_comment != doc_comment:
+                # Something changed, let's apply a result.
+                diff = Diff(file)
+
+                # Delete the old comment
+                diff.delete_lines(doc_comment.range.start.line,
+                                  doc_comment.range.end.line)
+
+                # Apply the new comment
+                assembled_string = new_comment.assemble()
+                assembled = assembled_string.splitlines(True)
+
+                end_line = new_comment.range.end.line
+                diff.add_lines(new_comment.range.start.line, assembled[:-1])
+                diff.add_lines(end_line, assembled[-1] +
+                               file[end_line - 1][len(assembled[-1]):])
```

Comment:
> hm I'm wondering why it does work... add_lines should only allow lists

you are right it thus breaks up the string because:
```
>>> x = "hello"
>>> list(x)
['h', 'e', 'l', 'l', 'o']
>>> 
```



> especially I'm wondering about end_line, as it's set to new_comment.range.end.line, and this is the same range like in the old doccomment^^

Even I am not sure what happens here :confused:, the old comment's range is always equal to the new_comment's range.